File: app/src/main/java/com/ps/simplepapersoccer/AI/RandomAI.java
package com.ps.simplepapersoccer.AI;

import com.ps.simplepapersoccer.AI.Abstraction.IGameAI;
import com.ps.simplepapersoccer.GameObjects.GameHandler;
import com.ps.simplepapersoccer.GameObjects.Move.PartialMove;
import com.ps.simplepapersoccer.GameObjects.Move.PossibleMove;

public class RandomAI implements IGameAI {
    public PartialMove MakeMove(GameHandler gameHandler)
    {
        PossibleMove randomMove = (PossibleMove)gameHandler.allPossibleMovesFromNode(gameHandler.ballNode).toArray()[0];
        return new PartialMove(randomMove.oldNode, randomMove.newNode, gameHandler.currentPlayersTurn);
    }
}

File: app/src/main/java/com/ps/simplepapersoccer/AI/randomAI/RandomAI.kt
package com.ps.simplepapersoccer.ai.randomAI

import com.ps.simplepapersoccer.ai.abstraction.IGameAI
import com.ps.simplepapersoccer.gameObjects.game.GameHandler
import com.ps.simplepapersoccer.gameObjects.move.PartialMove
import com.ps.simplepapersoccer.gameObjects.player.AIPlayer

class RandomAI(playerNumber: Int, playerColor: Int, playerName: String = RandomAI::class.java.simpleName): IGameAI, AIPlayer(playerName, playerNumber, playerColor) {
    override suspend fun makeMove(gameHandler: GameHandler): PartialMove {
        return PartialMove(gameHandler.ballNode,
                gameHandler.gameBoard.allLegalMovesFromBallNode.random().newNode,
                gameHandler.gameBoard.currentPlayersTurn)
    }
}